src/geometry/manifold/integral_curve.lean

```diff
@@ -0,0 +1,290 @@
+/-
+Copyright (c) 2022 Winston Yin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Winston Yin
+-/
+import analysis.ODE.picard_lindelof
+import geometry.manifold.cont_mdiff
+import geometry.manifold.mfderiv
+
+/-!
+# Integral curves of vector fields on a manifold
+
+For any continuously differentiable vector field on a manifold `M` and any chosen non-boundary point
+`x₀ : M`, an integral curve `γ : ℝ → M` exists such that `γ 0 = x₀` and the tangent vector of `γ` at
+`t` coincides with the vector field at `γ t` for all `t` within an open interval around 0.
+
+As a corollary, such an integral curve exists for any starting point `x₀` if `M` is a manifold
+without boundary.
+
+## Implementation details
+
+Since there is already an ODE solution existence theorem
+`ODE_solution_exists.at_ball_of_cont_diff_on_nhds`, the bulk of this file is to convert statements
+about manifolds to statements about the model space. This comes in a few steps:
+1. Express the smoothness of the vector field `v` in a single fixed chart around the starting point
+`x₀`.
+2. Use the ODE solution existence theorem to obtain a curve `γ : ℝ → M` whose derivative coincides
+with the vector field (stated in the local chart around `x₀`).
+3. Same as 2 but now stated in the local chart around `γ t`, which is how `cont_mdiff` is defined.
+
+## Tags
+
+integral curve, vector field
+-/
+
+localized "notation (name := ext_chart_at) `𝓔(` I `, ` x `)` :=
+  ext_chart_at I x" in manifold
+
+open_locale manifold
+
+namespace model_with_corners
+
+variables
+  {𝕜 : Type*} [nontrivially_normed_field 𝕜]
+  {E : Type*} [normed_add_comm_group E] [normed_space 𝕜 E]
+  {H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+  {M : Type*} [topological_space M] [charted_space H M]
+
+lemma boundaryless.is_open_target
+  [I.boundaryless] {x : M} : is_open (ext_chart_at I x).target :=
+begin
+  rw [ext_chart_at_target, model_with_corners.boundaryless.range_eq_univ, set.inter_univ],
+  exact (model_with_corners.continuous_symm _).is_open_preimage _ (local_homeomorph.open_target _)
+end
+
+/-- An interior point of a manifold is a point whose image in the model vector space is in the
+interior of the chart's target.
+
+TODO : This definition refers to a chosen chart at `x`, but the property is independent of the
+choice of chart. See Theorem 1.37 (for general manifolds) and Theorem 1.46 (for smooth manifolds) of
+Introduction to Smooth Manifolds by John M. Lee.
+-/
+def is_interior_point (x : M) := 𝓔(I, x) x ∈ interior 𝓔(I, x).target
+
+lemma boundaryless.is_interior_point
+  [I.boundaryless] {x : M} : I.is_interior_point x :=
+begin
+  rw [is_interior_point, is_open.interior_eq (boundaryless.is_open_target I)],
+  exact local_equiv.map_source _ (mem_ext_chart_source _ _)
+end
+
+end model_with_corners
```

Comment:
This is not specific to integral curves of vector fields. Could you move that to better places?